gnd/src/main/java/com/google/android/gnd/ui/recorddetails/RecordDetailsViewModel.java

```diff
@@ -17,32 +17,124 @@
 package com.google.android.gnd.ui.recorddetails;
 
 import android.arch.lifecycle.LiveData;
-import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.LiveDataReactiveStreams;
+import android.util.Log;
+import android.view.View;
+
 import com.google.android.gnd.repository.DataRepository;
 import com.google.android.gnd.repository.Resource;
 import com.google.android.gnd.ui.common.AbstractViewModel;
+import com.google.android.gnd.vo.Form;
 import com.google.android.gnd.vo.Record;
+import com.google.common.collect.ImmutableList;
+
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+import io.reactivex.subjects.BehaviorSubject;
+import io.reactivex.subjects.Subject;
+
 import javax.inject.Inject;
 
 public class RecordDetailsViewModel extends AbstractViewModel {
 
+  private static final String TAG = RecordDetailsViewModel.class.getSimpleName();
+
   private final DataRepository dataRepository;
-  private final MutableLiveData<Resource<Record>> record;
+  private final Subject<RecordDetailsFragmentArgs> recordDetailsRequest;
+  public final Flowable<Resource<Record>> recordStream;
+  public final LiveData<Resource<Record>> records;
+  public final LiveData<Integer> progressBarVisibility;
+  public final LiveData<String> toolbarTitle;
+  public final LiveData<String> toolbarSubtitle;
+  public final LiveData<String> formNameView;
 
   @Inject
   RecordDetailsViewModel(DataRepository dataRepository) {
     this.dataRepository = dataRepository;
-    this.record = new MutableLiveData<>();
+
+    this.recordDetailsRequest = BehaviorSubject.create();
+
+    this.recordStream =
+        recordDetailsRequest
+          .switchMap(a ->
+              this.dataRepository.getRecordDetails(a.getProjectId(), a.getFeatureId(), a.getRecordId()))
+          .toFlowable(BackpressureStrategy.LATEST);
+
+    //TODO: Refactor to expose the fetched record directly.
+    this.records = LiveDataReactiveStreams.fromPublisher(
+        recordStream
+    );
+
+    this.progressBarVisibility = LiveDataReactiveStreams.fromPublisher(
+        recordStream.map(this::toProgressBarVisibility)
+        .onErrorReturn(x -> View.GONE)
+    );
+
+    this.toolbarTitle = LiveDataReactiveStreams.fromPublisher(
+        recordStream.map(this::toToolbarTitle)
+        .onErrorReturn(x -> "")
+    );
+
+    this.toolbarSubtitle = LiveDataReactiveStreams.fromPublisher(
+        recordStream.map(this::toToolbarSubtitle)
+        .onErrorReturn(x -> "")
+    );
+
+    this.formNameView = LiveDataReactiveStreams.fromPublisher(
+        recordStream.map(this::toFormNameView)
+        .onErrorReturn(x -> "")
+    );
+
+    recordDetailsRequest.subscribe(args -> {
+      Log.d(TAG, "" + args);
+    });
+  }
+
+  public void getRecordDetailsRequest(RecordDetailsFragmentArgs args) {
+    this.recordDetailsRequest.onNext(args);
+  }
+
+  private Integer toProgressBarVisibility(Resource<Record> record) {
```

Comment:
I guess we need to return Integer instead of int because autoboxing doesn't work on lambdas (i.e. recordStream.map(this::toProgressBarVisibility)?